src/api/v1beta1/properties.go

```diff
@@ -487,3 +488,15 @@ func (dk *DynaKube) GetOneAgentEnvironment() []corev1.EnvVar {
 	}
 	return []corev1.EnvVar{}
 }
+
+func (dk *DynaKube) ShallUpdateOneAgentConnectionInfo(timeProvider *timeprovider.Provider) bool {
```

Comment:
pls, don't use "shall"", as its a not used anywhere in the codebase, and it shouldn't be anyway.

"Shall" is mainly used to express future events, or intentions (and not true/false questions). Furthermore (in my experience )its not really used in normal speech or text ("will" is waaay more common). Most common places I encountered "shall" were in, songs, videogames and fantasy novels. None of which fits our code 😄 .

There are way more generic, better and simpler options:
- `NeedsOneAgentConnectionInfoUpdate`
- `IsOutdatedOneAgentConnectionInfo`

Reply:
I don't like this "is outdated" phrase, that sounds yukky to me. I probably can (and hence will) life with "Needs..."

Reply:
In my opinion I agree with @0sewa0. - Never saw "Shall" in any code and gives me a hard time reading it. Would rather vote for `IsOutdatedOneAgentConnectionInfo` to aline with the other functions in our code. - "Is" fits perfect here as it describes the state if the struct it refers to.

Reply:
On second thought all of suggestions so far are wrong, because what this property says, that it is now allowed to do an update wrt. the timeout restrictions. It doesn't say if the update has to be done, if it needs to be done or of the connection info is already outdated. The dynakube instance at hand just doesn't know that.

Reply:
let's go with one of marcell's suggestions. we should avoid "shall" if the meaning is ambiguous and there are other, well-known words with the same meaning already used in the code base.